psi4/driver/p4util/python_helpers.py

```diff
@@ -93,18 +94,87 @@ def pybuild_wavefunction(mol, basis=None):
 ## Python JK helps
 
 @staticmethod
-def pybuild_JK(basis, aux=None):
+def pybuild_JK(orbital_basis, aux=None, jk_type=None):
+    """
+    Constructs a Psi4 JK object from an input basis.
+
+    Parameters
+    ----------
+    orbital_basis : :py:class:`~psi4.core.BasisSet`
+        Orbital basis to use in the JK object.
+    aux : :py:class:`~psi4.core.BasisSet`
+        Optional auxiliary basis set for density-fitted tensors. Defaults
+        to the DF_BASIS_SCF if set, otherwise the correspond JKFIT basis
+        to the passed in orbital_basis.
+    type : str
+        Type of JK object to build (DF, Direct, PK, etc). Defaults to the
+        current global SCF_TYPE option.
+
+    Returns
+    -------
+    :py:class:`~psi4.core.JK`
+        Uninitialized JK object.
+
+    Example
+    -------
+
+    jk = psi4.core.JK.build(bas)
+    jk.set_memory(int(5e8)) # 4GB of memory
+    jk.initialize()
+
+    ...
+
+    jk.C_left_add(matirx)
+    jk.compute()
+    jk.C_clear()
+
+    ...
+
+    """
+
+    optstash = optproc.OptionsState(["SCF_TYPE"])
+
+    if jk_type is not None:
+        core.set_global_option("SCF_TYPE", jk_type)
+
     if aux is None:
-        aux = core.BasisSet.build(basis.molecule(), "DF_BASIS_SCF",
-                                      core.get_option("SCF", "DF_BASIS_SCF"),
-                                      "JKFIT", core.get_global_option('BASIS'),
-                                      basis.has_puream())
+        if core.get_option("SCF", "SCF_TYPE") == "DF":
+            aux = core.BasisSet.build(basis.molecule(), "DF_BASIS_SCF",
+                                          core.get_option("SCF", "DF_BASIS_SCF"),
+                                          "JKFIT", core.get_global_option('BASIS'),
+                                          basis.has_puream())
+        else:
+            aux = core.BasisSet.zero_ao_basis_set()
 
-    return core.JK.build_JK(basis, aux)
+
+    jk = core.JK.build_JK(basis, aux)
+
+    optstash.restore()
+    return jk
 
 core.JK.build = pybuild_JK
 
 ## Python other helps
 
 core.Molecule.run_dftd3 = qcdb.interface_dftd3.run_dftd3
 
+
+def set_options(options_dict):
+    """
+    Sets Psi4 global options from an input dictionary.
+    """
+
+    for k, v, in options_dict.items():
+        core.set_global_option(k, v)
```

Comment:
I think should do `k.upper()`. Since options setting has always been within developer scope, it's not (necessarily) case insensitive. The inputparser does this, so should imitate that here, I think.